cmd/login/wc.go

```diff
@@ -90,52 +61,138 @@ func (r *runner) createClusterClientCert(ctx context.Context, client k8sclient.I
 		}
 		releaseService, err = release.New(serviceConfig)
 		if err != nil {
-			return microerror.Mask(err)
+			return serviceSet{}, microerror.Mask(err)
 		}
 	}
 
+	return serviceSet{
+		clientCertService:   clientCertService,
+		organizationService: organizationService,
+		clusterService:      clusterService,
+		releaseService:      releaseService,
+	}, nil
+}
+
+func (r *runner) getCluster(ctx context.Context, services serviceSet, provider string) (*cluster.Cluster, error) {
+	var err error
 	var c *cluster.Cluster
-	{
-		var namespaces []string
-		if len(r.flag.WCOrganization) > 0 {
-			orgNamespace, err := getOrganizationNamespace(ctx, organizationService, r.flag.WCOrganization)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-
-			namespaces = append(namespaces, orgNamespace)
-		} else {
-			namespaces, err = getAllOrganizationNamespaces(ctx, organizationService)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-		}
 
-		if r.flag.WCInsecureNamespace {
-			namespaces = append(namespaces, "default")
+	var namespaces []string
+	if len(r.flag.WCOrganization) > 0 {
+		orgNamespace, err := getOrganizationNamespace(ctx, services.organizationService, r.flag.WCOrganization)
+		if err != nil {
+			return nil, microerror.Mask(err)
 		}
 
-		c, err = findCluster(ctx, clusterService, organizationService, provider, r.flag.WCName, namespaces...)
+		namespaces = append(namespaces, orgNamespace)
+	} else {
+		namespaces, err = getAllOrganizationNamespaces(ctx, services.organizationService)
 		if err != nil {
-			return microerror.Mask(err)
+			return nil, microerror.Mask(err)
 		}
 	}
 
+	if r.flag.WCInsecureNamespace {
+		namespaces = append(namespaces, "default")
+	}
+
+	c, err = findCluster(ctx, services.clusterService, services.organizationService, provider, r.flag.WCName, namespaces...)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	return c, nil
+}
+
+func (r *runner) getCertOperatorVersion(c *cluster.Cluster, provider string, services serviceSet, ctx context.Context) (string, error) {
 	releaseVersion, err := getClusterReleaseVersion(c, provider, r.flag.WCInsecureNamespace)
```

Comment:
This line of code will still be executed for pure CAPI clusters (e.g. OpenStack), right? If so, OpenStack clusters don't have a release label so an error will be returned and this function will fail (i.e. won't get to line 115).

Reply:
Yes, this will fail if there is no release label. We also assume that the release CR exists.
AFAIK we currently also need the label to tell if a cluster is in CAPA/CAPZ.
It worked during testing, though. Do we have MCs without releases already?

Reply:
No OpenStack MCs should have Release CRs (see https://github.com/giantswarm/openstack-app-collection/pull/39) and no WCs should have release labels (removed in cluster-openstack v0.6.0 https://github.com/giantswarm/cluster-openstack/blob/main/CHANGELOG.md#060---2022-02-28).

Reply:
Alright.
It's no problem to get rid of the release version concept here for openstack. I think we can also get rid of the version validation completely since all azure and aws cluster should support client cert creation at this point.
To support CAPA and CAPZ we are going to need something else however, since they are currently identified through versioning.